apm-agent-core/src/main/java/co/elastic/apm/agent/metrics/builtin/SystemMetrics.java

```diff
@@ -78,26 +97,133 @@
+                            }
+                            foundRegex = applyCgroup1Regex(mountLine);
+                            if (foundRegex != null) {
+                                cgroupFilesTest = verifyCgroup1Available(new File(foundRegex));
+                                if (cgroupFilesTest != null) return cgroupFilesTest;
+                            }
+                        }
+                    }
+                    // Fall back to /sys/fs/cgroup if not found on mountinfo
+                    cgroupFilesTest = verifyCgroup2Available(lineCgroup, new File(SYS_FS_CGROUP));
+                    if (cgroupFilesTest != null) return cgroupFilesTest;
+                    cgroupFilesTest = verifyCgroup1Available( new File(SYS_FS_CGROUP + File.pathSeparator + "memory"));
+                    if (cgroupFilesTest != null) return cgroupFilesTest;
+
+                }
+            } catch (Exception e) {
+            }
+        }
+        return null;
+    }
+    String applyCgroup1Regex(String mountLine) {
+        Matcher matcher = CGROUP1_MOUNT_POINT.matcher(mountLine);
+        if (matcher.matches()) {
+            return matcher.group(1);
+        }
+        return null;
+    }
+    String applyCgroup2Regex(String mountLine) {
+        Matcher matcher = CGROUP2_MOUNT_POINT.matcher(mountLine);
+        if (matcher.matches()) {
+            return matcher.group(1);
+        }
+        return null;
+    }
+    private CgroupFiles verifyCgroup2Available(String lineCgroup, File mountDiscovered) throws IOException {
+        final String[] cgroupSplit = StringUtils.split(lineCgroup, ':');
+        // Checking cgroup2
+        File maxMemory = new File(mountDiscovered, cgroupSplit[cgroupSplit.length - 1] + "/" + CGROUP2_MAX_MEMORY);
+        if (maxMemory.canRead()) {
+            try(BufferedReader fileReaderMem = new BufferedReader(new FileReader(maxMemory))) {
+                String memMaxLine = fileReaderMem.readLine();
+                if (!"max".equalsIgnoreCase(memMaxLine)) {
+                    return new CgroupFiles(maxMemory,
+                        new File(mountDiscovered, cgroupSplit[cgroupSplit.length - 1] + "/" + CGROUP2_USED_MEMORY),
+                        new File(mountDiscovered, cgroupSplit[cgroupSplit.length - 1] + "/" + CGROUP2_STAT_MEMORY));
+                }
+            }
+        }
+        return null;
+    }
+
+    private CgroupFiles verifyCgroup1Available(File mountDiscovered) throws IOException {
+        // Checking cgroup1
+        File maxMemory = new File(mountDiscovered, CGROUP1_MAX_MEMORY);
+        if (maxMemory.canRead()) {
+            try(BufferedReader fileReaderMem = new BufferedReader(new FileReader(maxMemory))) {
+                String memMaxLine = fileReaderMem.readLine();
+                long memMax = Long.parseLong(memMaxLine);
+                if (memMax < UNLIMITED) { // Cgroup1 use a contant to disabled limits
```

Comment:
No need to look at unlimited

Reply:
I read about to track cgroup itens even if it is not limited. 
My afraid is that on normal use cases, like running on a linux environment these metrics be collected, and it will track all host values, like memory usage, it will be memory usage of all OS, because nothing is limited on cgroup.